src/functions-and-arrays.js

```diff
@@ -1,20 +1,172 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2){
+  return Math.max(num1, num2);
+}
+
 // Iteration #2: Find longest word
+// -- Declare a function named findLongestWord that takes as an argument an array of words and returns the longest one. 
+// -- If there are 2 with the same length, it should return the first occurrence.
+
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-// Iteration #3: Calculate the sum
+function findLongestWord(arrayWords) {
+  if (arrayWords.length === 0) {
+    return null;
+  }
+
+  // declaramos variables de control
+  let largura = 0;
+  let palabraMaslarga; // si let palabraMasLarga = null --> // significa que la variable no apunta a cap lloc de la memoria de la computadora
 
+  // loop para chequear todas las palabras
+  for(let i = 0; i < arrayWords.length; i++) {
+    if (arrayWords[i].length > largura) {
+      largura = arrayWords[i].length;
+      palabraMaslarga = arrayWords[i];
+    }
+  }
+
+  // retornamos palabra más larga
+  return palabraMaslarga;
+
+};
+
+// Iteration #3: Calculate the sum
+// -- Declare a function named sumNumbers that takes an array of numbers as an argument, and returns the sum of all of the numbers in the array.
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+// tenemos que sumar todos los valores del array
+
+function sumNumbers(arrayNumbers) {
+
+  // declaramos variables de control
+  let suma = 0;
+
+  // loop para chequear todos los números del array
+  for (let i = 0; i < arrayNumbers.length; i++) {
+    suma += arrayNumbers[i];
+  }
+
+  // retornamos valor de la suma
+  return suma;
+  
+};
+
+// BONUS
+// -- Create a new function sum() that calculates the sum for array filled with (almost) any type of data.
+// -- Strings should have their length added to the total.
+// -- Boolean values should be coerced into their corresponding numeric values
+
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+function sum(arrayItems){
+
+  // declaramos variables de control
+  let suma = 0;
+
+  // loop para chequear todos los números del array
+  for (let i = 0; i < arrayItems.length; i++){
+    // condicionales definir el valor de cada item en función de su data type
```

Comment:
Aca si te fijas bien el unico caso distinto es el de string por que le sumas el length. Entonces podrias hacer algo asi:

`
  for(let item of arrayItems) {
     if(typeof item === "string") {
       suma += item.length
    }  else {
    suma += item
  }

   
  }
`